lahja/endpoint.py

```diff
@@ -33,62 +41,104 @@
 )
 
 
+class ConnectionConfig(NamedTuple):
+    """
+    Configuration class needed to establish :class:`~lahja.endpoint.Endpoint` connections.
+    """
+    name: str
+    path: pathlib.Path
+
+    @staticmethod
+    def from_name(name: str) -> 'ConnectionConfig':
```

Comment:
You could give *optional* control over the base path by allowing a `Path` as an optional second argument.

```python
def from_name(name, base_path=None):
    if base_path is None:
        return ConnectionConfig(...)
    elif base_path.is_dir():
        return ConnectionConfig(name=name, path = base_path / name)
    else:
        raise TypeError("Something something bad path explanation")
```

Reply:
Looks useful :+1: